optimizely/client/client.go

```diff
@@ -364,43 +360,107 @@ func (o *OptimizelyClient) GetAllFeatureVariables(featureKey string, userContext
 	return enabled, variableMap, err
 }
 
-func (o *OptimizelyClient) getFeatureDecision(featureKey string, userContext entities.UserContext) (decisionContext decision.FeatureDecisionContext, featureDecision decision.FeatureDecision, err error) {
+// GetVariation returns the key of the variation the user is bucketed into
+func (o *OptimizelyClient) GetVariation(experimentKey string, userContext entities.UserContext) (result string, err error) {
+
+	defer func() {
+		if r := recover(); r != nil {
+			switch t := r.(type) {
+			case error:
+				err = t
+			case string:
+				err = errors.New(t)
+			default:
+				err = errors.New("unexpected error")
+			}
+			errorMessage := fmt.Sprintf("optimizely SDK is panicking with the error:")
+			logger.Error(errorMessage, err)
+			logger.Debug(string(debug.Stack()))
+		}
+	}()
+
+	_, experimentDecision, err := o.getExperimentDecision(experimentKey, userContext)
+	if err != nil {
+		logger.Error("received an error while computing experiment decision", err)
+	}
+
+	if experimentDecision.Variation != nil {
+		result = experimentDecision.Variation.Key
+	}
+
+	return result, err
+}
+
+// Track take and event key with event tags and if the event is part of the config, send to events backend.
+func (o *OptimizelyClient) Track(eventKey string, userContext entities.UserContext, eventTags map[string]interface{}) (err error) {
 
 	defer func() {
-		var e error
 		if r := recover(); r != nil {
 			switch t := r.(type) {
 			case error:
-				e = t
+				err = t
 			case string:
-				e = errors.New(t)
+				err = errors.New(t)
 			default:
-				e = errors.New("unexpected error")
+				err = errors.New("unexpected error")
 			}
 			errorMessage := fmt.Sprintf("optimizely SDK is panicking with the error:")
-			logger.Error(errorMessage, e)
+			logger.Error(errorMessage, err)
 			logger.Debug(string(debug.Stack()))
+		}
+	}()
+
+	projectConfig, err := o.GetProjectConfig()
+	if err != nil {
+		logger.Error("Optimizely SDK tracking error", err)
+		return err
+	}
+
+	configEvent, err := projectConfig.GetEventByKey(eventKey)
+
+	if err == nil {
```

Comment:
if possible, check for not nil error first  - it will eliminate else/if - and it will be cleaner

Reply:
ok